qt/widgets/plotting/src/Qwt/PreviewPlot.cpp

```diff
@@ -987,3 +987,11 @@ void PreviewPlot::handleAxisTypeSelect() {
   // Update the plot
   emit needToHardReplot();
 }
+
+/**
+ * Removes the Y-Axis option from the context menu as the Sample Transmission
+ * Calculator does not need it.
+ */
+void PreviewPlot::disableYAxisMenu() {
+  m_contextMenu->actions().at(4)->setVisible(false);
```

Comment:
This is not very robust - it will break if the order is ever changed for some reason. There's no unit tests for this whole thing, so it won't be detected at all, until someone looks at it.

It should use a more robust approach instead - setting the object name via `menu->setObjectName(<insert_string_here>);`, in the `addOptionsToMenus` function. The string should be a static class variable to make it easier to re-use. Also watch out as the function is used multiple times for different menus - the `menuName` should also be part of the object name!

The main menu `m_contextMenu` can then be queried for this specific object via `m_contextMenu->findChild<QMenu*>(same_name_as_above);`